src/main/java/org/apache/sling/feature/maven/mojos/ExtractExtensionMojo.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.sling.feature.maven.mojos;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.sling.feature.Artifact;
+import org.apache.sling.feature.Extension;
+import org.apache.sling.feature.Feature;
+import org.apache.sling.feature.maven.ProjectHelper;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Map;
+
+@Mojo(name = "extract-extension",
+    defaultPhase = LifecyclePhase.PACKAGE,
+    threadSafe = true)
+public class ExtractExtensionMojo extends AbstractFeatureMojo {
```

Comment:
I think we should not come up with a new way of selecting a feature file but use the existing mechanism that is used in other mojos (extending AbstractIncludingFeatureMojo and using FeatureSelectionConfig)

Reply:
Hi @cziegeler thanks for the feedback. 

The reason why I didn't use that is because the AbstractIncludingFeatureMojo allows the selection of feature file sets where here I think it makes most sense to work with a single feature file for the extract-extension functionality.

What I could do is work with the mechanism you suggest and then give an error if multiple feature files are selected. WDYT?

Reply:
That would be one way, yes. The other option would be to specify an "output directory" and the files generated are using the classifier and the extension name to create the file (and then either json or txt as extension)